src/bin/compute-wer.cc

```diff
@@ -91,36 +119,26 @@ int main(int argc, char *argv[]) {
       }
       num_words += ref_sent.size();
       int32 ins, del, sub;
-      word_errs += LevenshteinEditDistance(ref_sent, hyp_sent, &ins, &del, &sub);
+      int32 this_word_errs = LevenshteinEditDistance(ref_sent, hyp_sent, &ins, &del, &sub);
+      word_errs += this_word_errs;
       num_ins += ins;
       num_del += del;
       num_sub += sub;
 
       num_sent++;
-      sent_errs += (ref_sent != hyp_sent);
+      int32 this_sent_errs = (ref_sent != hyp_sent);
```

Comment:
Using a boolean as an int is a little unclear as it requires knowledge of
how C handles this.  Better to be more explicit, with a ? expression.

On Fri, May 27, 2016 at 5:41 PM, Vimal Manohar <notifications@github.com>
wrote:

> In src/bin/compute-wer.cc
> <https://github.com/kaldi-asr/kaldi/pull/777#discussion_r64970058>:
>
> >        num_ins += ins;
> >        num_del += del;
> >        num_sub += sub;
> >
> >        num_sent++;
> > -      sent_errs += (ref_sent != hyp_sent);
> > +      int32 this_sent_errs = (ref_sent != hyp_sent);
>
> I wasn't able to change this because this_sent_errs is required later in
> line 143:
> PrintWerResults(detailed_error_output.Stream(), ref_sent.size(), 1,
> this_word_errs, this_sent_errs, ins, del, sub,
> key + " ", 0);
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/777/files/6d65f2c860593df154d15d5f1440ac639ac9e23d#r64970058>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADJVu1JYKxlQJhPc02RE6Gj0Vm9CKRiwks5qF2TtgaJpZM4IeUC0>
> .
>